File: leetcode/python/find-missing-and-repeated-values.py
#!/usr/bin/env python3

""" Leetcode problem 2965: Find Missing and Repeated Values. """

from typing import List
from collections import defaultdict


class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        n = len(grid)
        count_map = defaultdict(int)
        a = 0
        b = 0
        for i in range(0, n):
            for j in range(0, n):
                count_map[grid[i][j]] += 1
        for i in range(1, (n * n) + 1):
            if count_map[i] == 2:
                a = i
            elif count_map[i] == 0:
                b = i
        return [a, b]


def main() -> None:
    pass


if __name__ == "__main__":
    main()

File: leetcode/rust/find-missing-and-repeated-values.rs
// Leetcode problem 2965: Find Missing and Repeated Values.

use std::collections::HashSet;

struct Solution {}

impl Solution {
    pub fn find_missing_and_repeated_values(grid: Vec<Vec<i32>>) -> Vec<i32> {
        let mut seen: HashSet<i32> = HashSet::new();
        let mut a = 0;
        let mut b = 0;
        for row in &grid {
            for entry in row {
                if seen.contains(&entry) {
                    a = *entry;
                } else {
                    seen.insert(*entry);
                }
            }
        }
        for i in 1..=grid.len() * grid.len() {
            if !seen.contains(&(i as i32)) {
                b = i as i32;
                break;
            }
        }
        [a, b].to_vec()
    }
}

fn main() {}
